src/vs/editor/contrib/rtv/RTVSynthDisplay.ts

```diff
@@ -118,173 +231,154 @@ export class RTVSynthDisplayBox implements IRTVDisplayBox{
 		return document.getElementById(this.getCellId(varname, idx)) as HTMLTableCellElement;
 	}
 
-	public async updateContent(allEnvs: any[], updateInPlace?: boolean, outputVars?: string[], prevEnvs?: Map<number, any>) {
+	// ------------
+	// front-end updates
+	// ------------
 
-		let outVarNames: string[];
-		if (!outputVars) {
-			outVarNames = [];
-		} else {
-			outVarNames = outputVars!;
-		}
+	public populateBoxContent(data: {[k: string]: [v: any]}) {
```

Comment:
This and `updateBoxContent` have very similar code. Can we merge them into a single method to avoid code duplication?